nixos/tests/discourse.nix

```diff
@@ -0,0 +1,34 @@
+import ./make-test-python.nix ({ pkgs, lib, ... }: {
+  name = "discourse";
+  meta.maintainers = with lib.maintainers; [ sorki ];
+
+  machine =
+    { ... }:
+    {
+      virtualisation.memorySize = 1024;
+      virtualisation.diskSize = 2 * 1024;
+      services.discourse = {
+        enable = true;
+        hostName = "localhost";
+        database = {
+          name = "discourse";
+          createLocally = true;
+        };
+        web = {
+          enable = true;
+          https = false;
+          workers = 1;
+        };
+      };
+    };
+
+  testScript = ''
+    start_all()
+
+    machine.wait_for_unit("discourse.service")
+
+    machine.sleep(60)
```

Comment:
Can it wait for the unicorn port instead? Or use `wait_until_succeeds` below?

Reply:
It can! Thanks!